tff_modular/modules/void/code/sub_modules/mob/void_mob.dm

```diff
@@ -0,0 +1,233 @@
+	light_blinking.Grant(src)
+
+	COOLDOWN_START(src, light_blink, VOID_LIGHT_BLINK_COOLDOWN)
+
+/mob/living/basic/void_creture/void_creture/Destroy()
+	. = ..()
+	void_scream.Destroy()
+	light_blinking.Destroy()
+	darknes_vision.Destroy()
+	teleport.Destroy()
+
+	if(GLOB.void_creature == src)
+		GLOB.void_creature = null
+
+/mob/living/basic/void_creture/Life(seconds_per_tick, times_fired)
+	. = ..()
+	if(isdead(src))
+		return
+
+	if(force_lighthing_blink && COOLDOWN_FINISHED(src, light_blink))
+		//В начале ищем окрудающие источники света и зставляем их тухнут/моргать.
+		for(var/atom/AO in range(9, src))
+			if(isturf(AO) || istype(AO, /obj/effect))
+				continue
+			if(!AO.light_on)
+				continue
+
+			if(istype(AO, /obj/machinery/light))
+				var/obj/machinery/light/l = AO
+				if(l.status == LIGHT_BROKEN)
+					continue
+				l.break_light_tube()
+
+			else if(istype(AO, /obj/item/flashlight))
+				var/obj/item/flashlight/f = AO
+				if(f.on)
+					f.make_flick()
+
+			else if(istype(AO, /obj/item/modular_computer))
+				var/obj/item/modular_computer/c = AO
+				c.make_flick()
+
+			//Самая сложная часть.. Вызывает эффекты у карбонов и их контента.
+			else if(istype(AO,	/mob/living/carbon))
+				var/mob/living/carbon/C = AO
+				//Вызываем эффект моргания у фонарика МОДа.
+				if(istype(C.back, /obj/item/mod/control))
+					var/obj/item/mod/control/mc = C.back
+					if(mc.active)
+						for(var/m in mc.modules)
+							if(istype(m, /obj/item/mod/module/flashlight))
+								var/obj/item/mod/module/flashlight/mf = m
+								mf.make_flick()
+
+				for(var/atom/thing in C.contents)
+					//Вызываем эффект моргания у ручного фонарика, если он есть.
+					if(istype(thing, /obj/item/flashlight))
+						var/obj/item/flashlight/f = thing
+						f.make_flick()
+						continue
+					//Вызываем эффект моргания у любых модульных устройств.
+					if(istype(thing, /obj/item/modular_computer))
+						var/obj/item/modular_computer/mc = thing
+						mc.make_flick()
+						continue
+					thing.set_light_on(FALSE)
+
+			else if(isobj(AO))
+			//Выключаем источник света... удачи с этим.
+				AO.set_light_on(FALSE)
+				AO.update_light()
+		COOLDOWN_START(src, light_blink, VOID_LIGHT_BLINK_COOLDOWN)
+
+	if(force_horror)
+		for(var/mob/living/carbon/human/h in view(src))
+			if(isdead(h) || !h.client || h.IsUnconscious())
+				continue
+			if(h.horror_state > HUMAN_HORROR_STATE_NORMAL)
+				continue
+			h.set_horror_state(HUMAN_HORROR_STATE_PANIC, time = 5 SECONDS)
+			h.add_screeen_temporary_effect(/atom/movable/screen/fullscreen/void_brightless)
+			h.light_out()
+
+/mob/living/basic/void_creture/melee_attack(atom/target, list/modifiers)
+	if(!combat_mode)
+		if(istype(target, /obj/machinery/door/airlock))
+			var/obj/machinery/door/airlock/a = target
+			playsound(a, 'sound/machines/airlock_alien_prying.ogg', 100, TRUE)
+
+			if(a.locked)
+				do_sparks(3, TRUE, a)
+				a.visible_message(span_black("[src.name], beggin corrupt [a.name] bolts with void!"))
```

Comment:
a.visible_message(span_userdanger("[src.name], starts melting through [a.name], slowly dissolving it in void!"))